### PR TITLE
Move from Travis to Github Actions

### PR DESCRIPTION
**Pull request by JoviDeCroock**
TODO:
- Migrate secrets from travis (done)
- Disable travis in settings (done)

Needs checking:
- Does coveralls upload correctly (seems good https://coveralls.io/builds/29228993)
- Does sauceLabs work correctly

Timings:
- Travis 2 * 50-78s (avg 60'ish)
- GH: 30 (build) + 50 (test)

Availability:
GH: 
image
Travis: https://www.traviscistatus.com/

### CODE REVIEW

**Comment by yisar** [comment]
What about icons in readme if Travis is removed? 
https://travis-ci.org/preactjs/preact.svg

**Comment by JoviDeCroock** [comment]
I mean, feel free to PR the icon, I consider them irrelevant since sauceLabs is basically our build icon

File: .github/workflows/ci-master.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - master
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2-beta
+      with:
+        fetch-depth: 1
```
Comment (developit) [inline .github/workflows/ci-master.yml]: I believe this is all good with the defaults in v2

```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/ci-master.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - run: npm ci
```
Comment (developit) [inline .github/workflows/ci-master.yml]: doesn't `npm ci` wipe `node_modules` before it runs?

I'm not 100% sure but I was fiddling around locally and it seems to do that.

If so, maybe we go with something like:

```sh
npm i --no-progress --no-fund --no-audit --prefer-offline
```

Comment (JoviDeCroock) [inline .github/workflows/ci-master.yml]:
```suggestion
    - run npm i --no-progress --no-fund --no-audit --prefer-offline
```

File: .github/workflows/ci-pr.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+name: CI-PR
+
+on: [pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2-beta
```
Comment (JoviDeCroock) [inline .github/workflows/ci-pr.yml]:
```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/ci-pr.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - run: npm ci
```
Comment (JoviDeCroock) [inline .github/workflows/ci-pr.yml]:
```suggestion
    - run: npm i --no-progress --no-fund --no-audit --prefer-offline
```

File: .github/workflows/ci-pr.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    - run: npm run build
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2-beta
```
Comment (JoviDeCroock) [inline .github/workflows/ci-pr.yml]:
```suggestion
    - uses: actions/checkout@v2
```

File: .github/workflows/ci-pr.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        # This uses the same name as the build-action so we can share the caches.
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - run: npm ci
```
Comment (JoviDeCroock) [inline .github/workflows/ci-pr.yml]:
```suggestion
    - run: npm i --no-progress --no-fund --no-audit --prefer-offline
```

File: .github/workflows/ci-master.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        cache-name: cache-node-modules
+      with:
+        path: ~/.npm
+        # This uses the same name as the build-action so we can share the caches.
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - run: npm ci
```
Comment (JoviDeCroock) [inline .github/workflows/ci-master.yml]:
```suggestion
    - run: npm i --no-progress --no-fund --no-audit --prefer-offline
```

File: .github/workflows/ci-master.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    - run: npm run build
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2-beta
```
Comment (JoviDeCroock) [inline .github/workflows/ci-master.yml]:
```suggestion
    - uses: actions/checkout@v2
```

**Pull request merged**